nsxt/resource_nsxt_policy_shared_resource.go

```diff
@@ -0,0 +1,207 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/terraform-provider-nsxt/api/infra/shares"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicySharedResource() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicySharedResourceCreate,
+		Read:   resourceNsxtPolicySharedResourceRead,
+		Update: resourceNsxtPolicySharedResourceUpdate,
+		Delete: resourceNsxtPolicySharedResourceDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicySharedResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"context":      getContextSchema(false, false, false),
+			"share_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Share ID to associate the resource to",
+				ForceNew:    true,
+			},
+			"resource_object": {
```

Comment:
This is plural in the API (resource_objects). Can we keep it plural here as well?

Reply:
We always use singular notation for a list of complex object like in this case, and a plural notation for lists of simple objects - this is due to the way that TF represents them. e.g.
```
   ip_addresses = ["1.1.1.1", "2.2.2.2"]
```

.vs.
```
   some_cidr_obj {
     addr = "1.1.1.0"
     prefix_len = "24"
  }
  some_cidr_obj {
     addr = "2.1.1.0"
     prefix_len = "24"
  }
```